[Fleet] Limit pagination size when retrieving full policy or withAgentCount

nchaulet opened the pull request:
## Description 

Resolve https://github.com/elastic/kibana/issues/192178

As retrieving agent policies with all the integration policy or with agent count could have a terrible performance impact, that PR aims to limit the pagination to 100 agent policies when retrieving policies with those options.

That PR also make a breaking change to the query parameters. Retrieving agent policies without agent count is now the new default behaviour and I introduced a new query parameter `withAgentCount` to retrieve the agent count

This will be a breaking change for 9.0

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/243896)
* Commit: 70872413c5e2c0ea8bce3c3039fefbc2c6233f30

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/243896#01929fe6-7959-457d-a668-f67530703234)

cc @nchaulet

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243896","number":243896,"commit":"70872413c5e2c0ea8bce3c3039fefbc2c6233f30"}],"number":243896}
buildkite-pr-comment-->

juliaElastic commented on x-pack/plugins/fleet/server/types/rest_spec/agent_policy.ts:
```diff
@@ -20,33 +20,42 @@ import { validateKuery } from '../../routes/utils/filter_utils';
+      ),
+    },
+    {
+      validate: (query) => {
```
Do we call the API anywhere from the UI where this could cause an issue?

nchaulet replied:
I just did a more in depth check and I fixed the few places where we were still calling that API without noAgentCount

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/243962)
* Commit: f3de12aef95d1dde7e09d9e176cb98871e86e91f

### Failed CI Steps
* [FTR Configs #3](https://buildkite.com/elastic/kibana-pull-request/builds/243962#0192a06e-c07a-48d2-bf66-cde1df780eb1)
* [FTR Configs #3](https://buildkite.com/elastic/kibana-pull-request/builds/243962#0192a08c-a93f-4312-a9ce-17dbc192f972)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/243962#0192a06e-c07a-48d2-bf66-cde1df780eb1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/243962/jobs/0192a06e-c07a-48d2-bf66-cde1df780eb1/artifacts/0192a08c-9513-4cec-a635-c6f65a24cba2) FTR Configs #<span></span>3 / Fleet Endpoints fleet policy secrets create package policy with secrets "after all" hook for "should return secret refs from agent policy API"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/243962#0192a08c-a93f-4312-a9ce-17dbc192f972) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/243962/jobs/0192a08c-a93f-4312-a9ce-17dbc192f972/artifacts/0192a099-3b89-46bf-b53a-f29f661c4c28) FTR Configs #<span></span>3 / Fleet Endpoints fleet policy secrets create package policy with secrets "after all" hook for "should return secret refs from agent policy API"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/243962#0192a06e-c07a-48d2-bf66-cde1df780eb1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/243962/jobs/0192a06e-c07a-48d2-bf66-cde1df780eb1/artifacts/0192a08c-9512-4dad-a772-d50c29acef19) FTR Configs #<span></span>3 / management Ingest pipelines Manage databases List returns existing databases

### History
* :broken_heart: [Build #243896](https://buildkite.com/elastic/kibana-pull-request/builds/243896) failed 70872413c5e2c0ea8bce3c3039fefbc2c6233f30

cc @nchaulet

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243962","number":243962,"commit":"f3de12aef95d1dde7e09d9e176cb98871e86e91f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243896","number":243896,"commit":"70872413c5e2c0ea8bce3c3039fefbc2c6233f30"}],"number":243962}
buildkite-pr-comment-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/244016)
* Commit: 7331d21b11707355d6248f66d9672ad79cef462f

### Failed CI Steps
* [Check OAS Snapshot](https://buildkite.com/elastic/kibana-pull-request/builds/244016#0192a0e8-2d71-49e9-95a1-95226f2f2e80)

### History
* :broken_heart: [Build #243962](https://buildkite.com/elastic/kibana-pull-request/builds/243962) failed f3de12aef95d1dde7e09d9e176cb98871e86e91f
* :broken_heart: [Build #243896](https://buildkite.com/elastic/kibana-pull-request/builds/243896) failed 70872413c5e2c0ea8bce3c3039fefbc2c6233f30

cc @nchaulet

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/244016","number":244016,"commit":"7331d21b11707355d6248f66d9672ad79cef462f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243962","number":243962,"commit":"f3de12aef95d1dde7e09d9e176cb98871e86e91f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243896","number":243896,"commit":"70872413c5e2c0ea8bce3c3039fefbc2c6233f30"}],"number":244016}
buildkite-pr-comment-->

nchaulet commented:
@juliaElastic I changed a little my PR to make that breaking less brutal/risky for users the GET /agent_policies` now do not retrieve agent count by default and I introduced a new `withAgentCount` parameter, I tested in depth our UI and nothing break.

juliaElastic commented on x-pack/plugins/fleet/server/types/rest_spec/agent_policy.ts:
```diff
@@ -20,33 +20,59 @@ import { validateKuery } from '../../routes/utils/filter_utils';
+      ),
+      noAgentCount: schema.maybe(
+        schema.boolean({
+          meta: { description: 'deprecated use withAgentCount instead' },
```
you can mark the field deprecated as `meta: { deprecated: true }`

nchaulet commented:
@elasticmachine merge upstream

nchaulet commented:
@elasticmachine merge upstream

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/244384)
* Commit: 7547eacd7235e6d9eef8fb8794093f1f775eb69d

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/46eda4c48922569954cdec2f7ac3facd60ab620c) | [after](https://github.com/elastic/kibana/commit/7547eacd7235e6d9eef8fb8794093f1f775eb69d) | diff |
| --- | --- | --- | --- |
| `fleet` | 1.7MB | 1.7MB | +2.0B |



### History
* :broken_heart: [Build #244349](https://buildkite.com/elastic/kibana-pull-request/builds/244349) failed 5998bfd2e0ffd4ede0f726f12f3d38babd6c0fd4
* :broken_heart: [Build #244016](https://buildkite.com/elastic/kibana-pull-request/builds/244016) failed 7331d21b11707355d6248f66d9672ad79cef462f
* :broken_heart: [Build #243962](https://buildkite.com/elastic/kibana-pull-request/builds/243962) failed f3de12aef95d1dde7e09d9e176cb98871e86e91f
* :broken_heart: [Build #243896](https://buildkite.com/elastic/kibana-pull-request/builds/243896) failed 70872413c5e2c0ea8bce3c3039fefbc2c6233f30

cc @nchaulet

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/244384","number":244384,"commit":"7547eacd7235e6d9eef8fb8794093f1f775eb69d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/244349","number":244349,"commit":"5998bfd2e0ffd4ede0f726f12f3d38babd6c0fd4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/244016","number":244016,"commit":"7331d21b11707355d6248f66d9672ad79cef462f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243962","number":243962,"commit":"f3de12aef95d1dde7e09d9e176cb98871e86e91f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243896","number":243896,"commit":"70872413c5e2c0ea8bce3c3039fefbc2c6233f30"}],"number":244384}
buildkite-pr-comment-->
